hack/generator/pkg/astmodel/property_definition.go

```diff
@@ -74,6 +77,27 @@ func (property *PropertyDefinition) WithValidation(validation Validation) *Prope
 	return &result
 }
 
+// WithoutValidation removes all validations from the field
+func (property *PropertyDefinition) WithoutValidation() *PropertyDefinition {
+	result := *property
+	result.validations = nil
+	return &result
+}
+
+// WithTag adds the given tag to the field
+func (property *PropertyDefinition) WithTag(key string, value string) *PropertyDefinition {
+	result := *property
+	// TODO: Should we have a copy function here to make this a bit safer? Right now both this function
+	// TODO: and the above WithValidations technically leave the reference the same.
```

Comment:
Should definitely respect the immutability contract for `PropertyDefinition` - but I think this line does the copy you're looking for:

``` go
result := *property
```

Reply:
My comment was mostly around the fact that `result := *property` is a shallow copy. So pointer fields aren't updated. Re-reading the comment I definitely didn't _say_ that very clearly but that is what I was getting at.

It's ok leaving pointer fields the same since they actually aren't changing, but it's just a little less "strong" of a copy that we could have -- we could implement a copy function where we created new ptrs too just to be safe (there's no functionality difference from what we have though).

Reply:
I've updated the TODO to be more clear at what I was getting at, at least temporarily.

I think before I merge this PR I should either:
1. Remove this TODO, and maybe rephrase as a note that just is an FYI - we aren't copying ptr fields but it's ok because we're immutable.
2. Add a `copy` function that does a deep copy by default and switch to use that.

Reply:
A `copy()` function sounds like a good idea.

Reply:
I've got this but will include it in a follow up PR (alongside some more JSON tags stuff)